### PR TITLE
Disconnect AppStatus from Stdout and allow writing to unix domain socket

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
`DaemonApp` now allows program to register the AppStatus instance
to which unix domain socket will be connected when `show_appstatus`
command is issued via `registerAppStatus` command.

Blocked on #538

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
Rebase please :)

**Comment by nemanja-boric-sociomantic** [comment]
Rebased :-)

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by mihails-strasuns-sociomantic** [comment]
@gavin-norman-sociomantic can you have a quick look too?

**Comment by gavin-norman-sociomantic** [comment]
Looking.

File: src/ocean/io/console/AppStatus.d [hunk 1]
```diff
@@ -287,6 +289,35 @@ public class AppStatus
 
     private int old_terminal_size;
 
+    /***************************************************************************
+
+        Output stream to output the streaming lines to. Can be null, in which
+        case output will be truncated.
+
+    ***************************************************************************/
+
```
Comment (gavin-norman-sociomantic) [inline src/ocean/io/console/AppStatus.d]: :grimacing: extra whitespace! :scream:

File: src/ocean/io/console/AppStatus.d [hunk 2]
```diff
@@ -287,6 +289,35 @@ public class AppStatus
...
+
+    private OutputStream stream;
+
+    /***************************************************************************
+
+        TerminalOutput stream to output the static lines to. Can be null, in
+        which case output will be truncated
+            .
```
Comment (gavin-norman-sociomantic) [inline src/ocean/io/console/AppStatus.d]: `.` ? :D

File: src/ocean/io/console/AppStatus.d [hunk 3]
```diff
@@ -287,6 +289,35 @@ public class AppStatus
...
+
+    ***************************************************************************/
+
+
+    private OutputStream stream;
+
+    /***************************************************************************
+
+        TerminalOutput stream to output the static lines to. Can be null, in
+        which case output will be truncated
```
Comment (gavin-norman-sociomantic) [inline src/ocean/io/console/AppStatus.d]: "truncated" implies that the lines will be output but cut off at a certain point. Is that the correct meaning? (I would have thought "disabled" would be more fitting.)

File: src/ocean/io/console/AppStatus.d [hunk 4]
```diff
@@ -301,11 +332,19 @@ public class AppStatus
                     title line
             ms_between_calls = expected milliseconds between calls to
                                getCpuUsage (defaults to 1000ms)
+            stream = stream to write into. Can be null if it's not yet
```
Comment (gavin-norman-sociomantic) [inline src/ocean/io/console/AppStatus.d]: Mention what's output to each destination.

File: src/ocean/io/console/AppStatus.d [hunk 5]
```diff
@@ -301,11 +332,19 @@ public class AppStatus
...
+                     available
+            terminal_output = terminal to write into. Can be null if it's not
+                     yet available
+            is_redirected = indicator if the output is redirected to a file
```
Comment (gavin-norman-sociomantic) [inline src/ocean/io/console/AppStatus.d]: Try to hide this internally, if it's only needed for the case of redirecting console output to a file.

**Review by gavin-norman-sociomantic — CHANGES_REQUESTED** [review]

**Comment by nemanja-boric-sociomantic** [comment]
Updated.

File: src/ocean/util/app/DaemonApp.d [hunk 6]
```diff
@@ -493,6 +534,74 @@ public abstract class DaemonApp : Application,
...
+        Registers AppStatus instance with the daemon application. This allows
+        DaemonApp to automatically connect unix domain socket to the AppStatus'
+        output.
+
+        Params:
+            app_status = instance of AppStatus to connect the output to.
+
+    **************************************************************************/
+
+    public void registerAppStatus (AppStatus app_status)
```
Comment (gavin-norman-sociomantic) [inline src/ocean/util/app/DaemonApp.d]: Would there be a way to reverse this? Have a `registerUnixSocket` method in `AppStatus`. It's a bit nasty to integrate this deprecated stuff into `DaemonApp`.

Comment (nemanja-boric-sociomantic) [reply src/ocean/util/app/DaemonApp.d]: Good point.

**Comment by nemanja-boric-sociomantic** [comment]
Updated.

**Review by gavin-norman-sociomantic — APPROVED** [review]
That looks better!

**Comment by nemanja-boric-sociomantic** [comment]
I think so as well, thanks for the comments!

**Pull request merged**